config/api/routes/users.php

```diff
@@ -184,7 +184,22 @@ function ($source, $filename) use ($id) {
 		],
 		'method'  => 'PATCH',
 		'action'  => function (string $id) {
-			return $this->user($id)->changePassword($this->requestBody('password'));
+			$user = $this->user($id);
+
+			// validate password of acting user unless they have logged in to reset it;
+			// always validate password of acting user when changing password of other users
+			if ($this->session()->get('kirby.resetPassword') !== true || $user->isLoggedIn() !== true) {
```

Comment:
`Suggestion:` My brain had troubles comprehend this. Mainly the second part. Is that to check whether the current user is `$user`? If so maybe `$this->user()->is($user)` could be the more easily parsable alternative?